timemachine/potentials/__init__.py

```diff
@@ -0,0 +1,144 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Callable, NamedTuple, Optional, Protocol, Sequence, Tuple
+
+import importlib_resources as resources
+import jax
+import numpy as np
+
+from timemachine.lib import custom_ops
+from timemachine.potentials import nonbonded
+
+Array = Any  # see https://github.com/google/jax/issues/943
+
+
+class Precision(Enum):
```

Comment:
The main reason was to be able to give a type to the `precision` arguments (I couldn't figure out a nice type e.g. for `np.float32`). I'm on the fence about this though; it would require updating in a lot of places.